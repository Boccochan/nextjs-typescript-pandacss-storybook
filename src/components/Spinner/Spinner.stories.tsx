import type { Meta, StoryObj } from "@storybook/react";

import { css } from "#/styled-system/css";

import { Spinner } from "./Spinner";

const meta = {
  title: "components/Spinner",
  component: Spinner,
  parameters: {
    layout: "centered",
  },
  decorators: [
    (Story) => (
      <div
        className={css({
          padding: "4",
          bg: "{colors.blue.500}",
          opacity: "0.7",
        })}
      >
        <Story />
      </div>
    ),
  ],
} satisfies Meta<typeof Spinner>;

export default meta;
type Story = StoryObj<typeof Spinner>;

export const Normal: Story = {};
