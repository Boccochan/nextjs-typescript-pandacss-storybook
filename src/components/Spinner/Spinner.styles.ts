import { cva, type RecipeVariantProps } from "#/styled-system/css";

export const spinner = cva({
  base: {
    fontSize: "6.8px",
    position: "relative",
    borderTop: "0.8em solid rgba(255, 255, 255, 0.2)",
    borderRight: "0.8em solid rgba(255, 255, 255, 0.2)",
    borderBottom: "0.8em solid rgba(255, 255, 255, 0.2)",
    borderLeft: "0.8em solid white",
    transform: "translateZ(0)",
    animation: "spinner 1.1s infinite linear",
    borderRadius: "50%",
    width: "3em",
    height: "3em",
  },
});

export type SpinnerVariants = NonNullable<RecipeVariantProps<typeof spinner>>;
