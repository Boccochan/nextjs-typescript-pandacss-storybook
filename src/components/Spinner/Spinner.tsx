import { spinner } from "./Spinner.styles";

export const Spinner = () => (
  <div className={spinner} role="alert" aria-label="loading" />
);
