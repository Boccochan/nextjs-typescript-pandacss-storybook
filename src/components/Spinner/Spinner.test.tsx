import { render } from "@testing-library/react";
import React from "react";

import { Spinner } from "./Spinner";

describe("Snapshot spinner", () => {
  it("Size=sm", async () => {
    const spinner = render(<Spinner size="sm" />);

    expect(spinner).toMatchSnapshot();
  });

  it("Size=md", async () => {
    const spinner = render(<Spinner size="md" />);

    expect(spinner).toMatchSnapshot();
  });

  it("Size=lg", async () => {
    const spinner = render(<Spinner size="lg" />);

    expect(spinner).toMatchSnapshot();
  });

  it("Size=xl", async () => {
    const spinner = render(<Spinner size="xl" />);

    expect(spinner).toMatchSnapshot();
  });
});
