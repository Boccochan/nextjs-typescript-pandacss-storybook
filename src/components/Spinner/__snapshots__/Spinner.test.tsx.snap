// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot spinner Size=lg 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-label="loading"
        class="fs_8.8px pos_relative border-t_0.8em_solid_rgba(255,_255,_255,_0.2) border-r_0.8em_solid_rgba(255,_255,_255,_0.2) border-b_0.8em_solid_rgba(255,_255,_255,_0.2) border-l_0.8em_solid_white transform_translateZ(0) animation_spinner_1.1s_infinite_linear rounded_50% w_3em h_3em"
        role="alert"
      />
    </div>
  </body>,
  "container": <div>
    <div
      aria-label="loading"
      class="fs_8.8px pos_relative border-t_0.8em_solid_rgba(255,_255,_255,_0.2) border-r_0.8em_solid_rgba(255,_255,_255,_0.2) border-b_0.8em_solid_rgba(255,_255,_255,_0.2) border-l_0.8em_solid_white transform_translateZ(0) animation_spinner_1.1s_infinite_linear rounded_50% w_3em h_3em"
      role="alert"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Snapshot spinner Size=md 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-label="loading"
        class="fs_6.8px pos_relative border-t_0.8em_solid_rgba(255,_255,_255,_0.2) border-r_0.8em_solid_rgba(255,_255,_255,_0.2) border-b_0.8em_solid_rgba(255,_255,_255,_0.2) border-l_0.8em_solid_white transform_translateZ(0) animation_spinner_1.1s_infinite_linear rounded_50% w_3em h_3em"
        role="alert"
      />
    </div>
  </body>,
  "container": <div>
    <div
      aria-label="loading"
      class="fs_6.8px pos_relative border-t_0.8em_solid_rgba(255,_255,_255,_0.2) border-r_0.8em_solid_rgba(255,_255,_255,_0.2) border-b_0.8em_solid_rgba(255,_255,_255,_0.2) border-l_0.8em_solid_white transform_translateZ(0) animation_spinner_1.1s_infinite_linear rounded_50% w_3em h_3em"
      role="alert"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Snapshot spinner Size=sm 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-label="loading"
        class="fs_4.8px pos_relative border-t_0.8em_solid_rgba(255,_255,_255,_0.2) border-r_0.8em_solid_rgba(255,_255,_255,_0.2) border-b_0.8em_solid_rgba(255,_255,_255,_0.2) border-l_0.8em_solid_white transform_translateZ(0) animation_spinner_1.1s_infinite_linear rounded_50% w_3em h_3em"
        role="alert"
      />
    </div>
  </body>,
  "container": <div>
    <div
      aria-label="loading"
      class="fs_4.8px pos_relative border-t_0.8em_solid_rgba(255,_255,_255,_0.2) border-r_0.8em_solid_rgba(255,_255,_255,_0.2) border-b_0.8em_solid_rgba(255,_255,_255,_0.2) border-l_0.8em_solid_white transform_translateZ(0) animation_spinner_1.1s_infinite_linear rounded_50% w_3em h_3em"
      role="alert"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Snapshot spinner Size=xl 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-label="loading"
        class="fs_10.8px pos_relative border-t_0.8em_solid_rgba(255,_255,_255,_0.2) border-r_0.8em_solid_rgba(255,_255,_255,_0.2) border-b_0.8em_solid_rgba(255,_255,_255,_0.2) border-l_0.8em_solid_white transform_translateZ(0) animation_spinner_1.1s_infinite_linear rounded_50% w_3em h_3em"
        role="alert"
      />
    </div>
  </body>,
  "container": <div>
    <div
      aria-label="loading"
      class="fs_10.8px pos_relative border-t_0.8em_solid_rgba(255,_255,_255,_0.2) border-r_0.8em_solid_rgba(255,_255,_255,_0.2) border-b_0.8em_solid_rgba(255,_255,_255,_0.2) border-l_0.8em_solid_white transform_translateZ(0) animation_spinner_1.1s_infinite_linear rounded_50% w_3em h_3em"
      role="alert"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
